### ISSUE TITLE
Serialize of Liste of Date does not follow pattern defined in @JsonFormat

### ISSUE THREAD

**Issue by camory** [vavr-io/vavr-jackson#154]
Hello,
I am facing the same issue as described in #141 but with List.
I have a DTO with a List `java.util.Date` with a custom date format:
```java
    @JsonFormat(pattern = "yyyy-MM-dd", timezone = "Europe/Paris")
    private final List<Date> dates;
```
When serializing, the date format is not taken into account.
It works when  replacing the io.vavr.collections.List by java.util.List.

**Comment by mincong-h** [comment]
Thanks for creating this issue, @camory . I confirm that this is a bug. I'm working on a fix.

**Comment by mincong-h** [comment]
@camory , I have a fix for your issue, see  #155

However, after the fix, I think you still need to modify your code to make it works. You need to provide the shape:

```diff
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "Europe/Paris")
-    @JsonFormat(pattern = "yyyy-MM-dd", timezone = "Europe/Paris")
     private final List<Date> dates;
```
